backend/app/controllers/spree/admin/stock_items_controller.rb

```diff
@@ -52,6 +52,7 @@ def load_stock_management_data
       def variant_scope
         scope = Spree::Variant.accessible_by(current_ability, :read)
         scope = scope.where(product: @product) if @product
+        scope = scope.order(:sku)
```

Comment:
Had to introduce this to make the spec deterministic. (The bug only manifests when the row which does not track inventory appears in the list *after* the row which doesn't.) AFAICS, there there is no pre-existing supposition about the ordering, so this shouldn't break anything (maybe a slight improvement in itself).